harness/determined/common/experimental/user.py

```diff
@@ -0,0 +1,93 @@
+import uuid
+from typing import Any, Dict, List, Optional
+
+from requests import Response
+
+from determined.common import api
+from determined.common.api import authentication, bindings
+
+
+# All the creates should be in
+class AgentUserGroup:
+    def __init__(
+        self,
+        agent_uid: Optional[int],
+        agent_gid: Optional[int],
+        agent_user: Optional[str],
+        agent_group: Optional[str],
+    ):
+        self.agent_uid = agent_uid
+        self.agent_gid = agent_gid
+        self.agent_user = agent_user
+        self.agent_group = agent_group
+
+
+class User:
+    def __init__(
+        self,
+        user_id: int, 
+        username: str,
+        password: str,
+        admin: bool,
+        session: api.Session,
+    ):
+        self.username = username
+        self.password = password
+        self.admin = admin
+        self.user_id = user_id
+        self.active = True
+        self.agent_uid = None
+        self.agent_gid = None
+        self.agent_user = None
+        self.agent_group = None
+        self.session = session
+
+    def update_user(
```

Comment:
I'm not a fan of having two ways to do the same thing in one API.

I'm not opposed to a single `user.update(username=..., password=...)`, etc, but I don't think we should also have `user.update_username()` and `user.update_password()`.

Reply:
I agree we can get rid of `update_username`.
we have `det user change-password` on the CLI, so maybe for the sake of alignment we could have `def change_password` here as well as a helpful shortcut?

Reply:
Changing update_username to rename because that's what the CLI has.